src/app/utils/researchNowUtils.js

```diff
@@ -0,0 +1,111 @@
+/*
+  `createResearchNowQuery()` currently handles:
+  `search_scope`, `dateAfter`, `dateBefore`,
+  `contributorLiteral`, `subjectLiteral`, `language`,
+  sorting, and pagination
+*/
+
+/* eslint-disable camelcase */
+
+const mapSearchScope = {
+  all: 'keyword',
+  contributor: 'author',
+  standard_number: 'standardNumber',
+  title: 'title',
+  date: 'date',
+};
+
+const filterObj = (field, value) => ({ field, value });
+const queryObj = (field, query) => ({ field, query });
+
+const mapFilters = (filters = {}) => {
+  const researchNowFilters = [];
+
+  const years = {};
+  if (filters.dateAfter) years.start = filters.dateAfter;
+  if (filters.dateBefore) years.end = filters.dateBefore;
+  if (years.start || years.end) researchNowFilters.push(filterObj('years', years));
+  if (filters.language) {
+    const languages = filters.language.map(lang => lang.replace('lang:', ''));
+    languages.forEach(language => researchNowFilters.push(filterObj('language', language)));
+  }
+
+  return researchNowFilters;
+};
+
+const createResearchNowQuery = (params) => {
```

Comment:
I think this could be made to handle GET and POST requests by adding an extra argument here that is set to either GET or POST.

If its POST, this acts as is, if its GET, instead of returning that object return a string created by something like:
```
const paramEncoder = (query) => {
  return Object.entries(query).map([key, value], => {
    let urlValue = `${key}=${value}`
    try {
      urlValue = `${key}=${JSON.stringify(value}`
    } catch (err) {
      // pass. If this fails it means the value is not JSON and we can take the initial value set above
    }
    return urlValue
  }).join('&');
}
```

Then I think you just URL encode that string and should be good to go?

Reply:
Or I guess this should default to POST, and change to GET if set, would make it so you have to change this call in fewer places.